### PR TITLE
Assignment4-Hadi_Cynthia

### PR DESCRIPTION
**Pull request by HadiM2**
@HadiM2
@montalcy

Site Link: https://zealous-sky-020793910.5.azurestaticapps.net/

### CODE REVIEW

**Comment by SergeiAshikhmin** [comment]
I will review this :)

**Review by SergeiAshikhmin — COMMENTED** [review]
The overall project looks good and the code is clean. Good job 👍 
I left comments here.

Features
Make all pages responsive ❌

1080p desktop ✅
Looks good on desktop

iPad Air ❌
Samsung Galaxy S20 Ultra Super Mega Excellent ❌
IPhone SE ❌
Every screen size smaller than desktop has bugs. 
Bugs:
- The sidebar is not visible. You can drage it but you have to know that it is there. Some menu icons on the top left corner would definitely help. 
- The timer in both games is missing.
-  Page size is wrong if the page is loaded in a small screen size. But the page is responsive if resized in the debug menu. Add a watch() to apply proper sizing.

Create a improved landing page and site theme (Background, make it look attractive to play) ✅
Great job, I really like the theme and neon. I am not sure if you implemented a background. 

Theme should extend all pages ✅

Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. 
The page should indicate Wordle of the Day or something like that ❌
You have the page but I didn't see any date picker to chosse a game.


Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌
I marked it wrong because there are no API calls, only hard coded data for the table. But the page looks good. 

It should indicate if the user has played that day ❌
Also didn't find. 

Clicking on the item should take the player to that game (Extra Credit) ❌

**Comment by MattA41** [comment]
I will review as well

File: wordle-web/pages/statistics.vue [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      prepend-icon="mdi-home-export-outline"
+      @click="router.push('/')"
+      color="secondary"
+      >Home</v-btn>
+      </div>
+    </v-card>
+  </div>
+</template>
+
+<script setup lang="ts">
```
Comment (MattA41) [inline wordle-web/pages/statistics.vue:49]: there is no API call here

File: wordle-web/components/Keyboard.vue [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
       cols="auto"
       v-for="letter in keyboardRow"
       :key="letter.char"
       :class="'ml-1 px-0'"
     >
-      <LetterResult :letter="letter" :clickable="true" />
+      <LetterResult :letter="letter" :clickable="true" style="color: blue;"/>
     </v-col>
   </v-row>
 </template>
```
Comment (MattA41) [inline wordle-web/components/Keyboard.vue:19]: you might want to take a look at doing some kind of refresh so the device does not have to rotate for the keyboard to display correctly

**Review by MattA41 — COMMENTED** [review]
Make all pages responsive ❌ Can I give 1/2 points? you have the same issue mine does where it wont be reactive till you rotate the device, but after the rotate its fine
  - 1080p desktop ✅
  - iPad Air ❌
  - Samsung Galaxy S20 Ultra Super Mega Excellent ❌
  - IPhone SE ❌
- Create a improved landing page and site theme (Background, make it look attractive to play) ✅
  - Theme should extend all pages ✅
- Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ❌✅
- Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ✅
- Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌ you are not actually calling the API just creating a table with fake data
  - It should indicate if the user has played that day ❌
  - Clicking on the item should take the player to that game (Extra Credit) ❌

File: Wordle.Api/Wordle.Api/Controllers/WordController.cs [hunk 3]
```diff
@@ -6,22 +6,21 @@ namespace Wordle.Api.Controllers;
...
+	/// </summary>
+	/// <param name="offsetInHours">Timezone offset in hours. Default to PST</param>
+	/// <returns></returns>
+	[HttpGet("WordOfTheDay")]
+	public async Task<string> GetWordOfDay(double offsetInHours = -7.0)
+	{
+		DateOnly today = DateOnly.FromDateTime(DateTime.UtcNow.AddHours(offsetInHours));
+		return await wordOfTheDayService.GetWordOfTheDayAsync(today);
+	}
+}
```
Comment (SaaWest) [inline Wordle.Api/Wordle.Api/Controllers/WordController.cs:26]: hint for word by date api call `[HttpGet("WordByDate/{month}/{day}/{year}")]
    public async Task<string?> GetWordByDate(int month, int day, int year)`

File: wordle-web/pages/statistics.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+const router = useRouter();
+//fake data 
+const tableData = ref([
```
Comment (SaaWest) [inline wordle-web/pages/statistics.vue:52]: should be able to quickly implement a call to the api.

**Review by SaaWest — COMMENTED** [review]
- Make all pages responsive ❌✅(note: there is no way to navigate between pages in other screen sizes other than desktop)
  - 1080p desktop ✅
  - iPad Air ❌
  - Samsung Galaxy S20 Ultra Super Mega Excellent ❌
  - IPhone SE ❌
- Create a improved landing page and site theme (Background, make it look attractive to play) ✅
  - Theme should extend all pages ❌✅ (I don't know if the picture from the landing should be extended to the other pages)
- Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
- Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ❌✅ (50/50 Cannot pick word from specific date)
- Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌✅ (no API call)
  - It should indicate if the user has played that day ❌ (need to indicate player)
  - Clicking on the item should take the player to that game (Extra Credit) ❌
  
  ## Extra Credit

- Clicking on an item in the daily word stats page should take the player to that game and allow them to play it ❌
- Create another ad page and choose one at random ❌

File: wordle-web/pages/statistics.vue (hunk 1 above)
Comment (montalcy) [reply wordle-web/pages/statistics.vue:49]: We're still working on it, our last database didn't work very well so we have been trying to figure it out here as well.

File: wordle-web/components/Keyboard.vue (hunk 2 above)
Comment (montalcy) [reply wordle-web/components/Keyboard.vue:19]: that's a good idea, hadn't thought of that

**Comment by HadiM2** [comment]
> The overall project looks good and the code is clean. Good job 👍 I left comments here.
> 
> Features Make all pages responsive ❌
> 
> 1080p desktop ✅ Looks good on desktop
> 
> iPad Air ❌ Samsung Galaxy S20 Ultra Super Mega Excellent ❌ IPhone SE ❌ Every screen size smaller than desktop has bugs. Bugs:
> 
> * The sidebar is not visible. You can drage it but you have to know that it is there. Some menu icons on the top left corner would definitely help.
> * The timer in both games is missing.
> * Page size is wrong if the page is loaded in a small screen size. But the page is responsive if resized in the debug menu. Add a watch() to apply proper sizing.
> 
> Create a improved landing page and site theme (Background, make it look attractive to play) ✅ Great job, I really like the theme and neon. I am not sure if you implemented a background.
> 
> Theme should extend all pages ✅
> 
> Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅ Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ❌ You have the page but I didn't see any date picker to chosse a game.
> 
> Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌ I marked it wrong because there are no API calls, only hard coded data for the table. But the page looks good.
> 
> It should indicate if the user has played that day ❌ Also didn't find.
> 
> Clicking on the item should take the player to that game (Extra Credit) ❌

Hey Sergei, thank you for the review! For the responsiveness on the required devices, it does actually work, it's a swipe gesture on the side to pull out the menu.

File: Wordle.Api/Wordle.Api/Controllers/WordController.cs (hunk 3 above)
Comment (HadiM2) [reply Wordle.Api/Wordle.Api/Controllers/WordController.cs:26]: Thanks Spencer! Will be implementing it very soon

File: wordle-web/pages/statistics.vue (hunk 4 above)
Comment (HadiM2) [reply wordle-web/pages/statistics.vue:52]: Almost there.

**Comment by HadiM2** [comment]
> * Make all pages responsive ❌✅(note: there is no way to navigate between pages in other screen sizes other than desktop)
>   
>   * 1080p desktop ✅
>   * iPad Air ❌
>   * Samsung Galaxy S20 Ultra Super Mega Excellent ❌
>   * IPhone SE ❌
> * Create a improved landing page and site theme (Background, make it look attractive to play) ✅
>   
>   * Theme should extend all pages ❌✅ (I don't know if the picture from the landing should be extended to the other pages)
> * Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
> * Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ❌✅ (50/50 Cannot pick word from specific date)
> * Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌✅ (no API call)
>   
>   * It should indicate if the user has played that day ❌ (need to indicate player)
>   * Clicking on the item should take the player to that game (Extra Credit) ❌
>   
>   ## Extra Credit
> * Clicking on an item in the daily word stats page should take the player to that game and allow them to play it ❌
> * Create another ad page and choose one at random ❌

Hey Spencer, thank you for the review! For the responsiveness on the required devices, it does actually work, it's a swipe gesture on the side to pull out the menu but in hindsight we should have something saying that! haha

**Comment by HadiM2** [comment]
> Make all pages responsive ❌ Can I give 1/2 points? you have the same issue mine does where it wont be reactive till you rotate the device, but after the rotate its fine
> 
> * 1080p desktop ✅
> * iPad Air ❌
> * Samsung Galaxy S20 Ultra Super Mega Excellent ❌
> * IPhone SE ❌
> * Create a improved landing page and site theme (Background, make it look attractive to play) ✅
>   
>   * Theme should extend all pages ✅
> * Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ❌✅
> * Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ✅
> * Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌ you are not actually calling the API just creating a table with fake data
>   
>   * It should indicate if the user has played that day ❌
>   * Clicking on the item should take the player to that game (Extra Credit) ❌

Hey Matt, thank you for the review! For the responsiveness on the required devices, it does actually work, it's a swipe gesture on the side to pull out the menu. Thank you for the note about the keyboard, we didn't know about that bug. Oh, also the instructions page is there, can be accessed by the Hamburger menu or a button on the homepage.

**Comment by MikaellaCroskrey-IntelliTect** [comment]
## Assignment 4
1. Make all pages responsive ✅ (13 out of 15 points)
  - 1080p desktop ✅
  - iPad Air ✅
  - Samsung Galaxy S20 Ultra Super Mega Excellent ✅ 
  - IPhone SE ✅
  Point deductions due to: The timer starts getting cut off when the width < 1280 and isn't visible at all on smaller screens. The sidebar is accessible via swipe on mobile screens but is difficult to find. 
  
2. Create an improved landing page and site theme (Background, make it look attractive to play) ✅
  - Theme should extend all pages ✅

3. Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅

4. Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ✅ (12 of 15 points) The WordOfTheDay page is hitting an error on the test site. Partial credit given for the code.

5. Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌ (3 of 10 points) Page exists but it only shows hardcoded data.
  - It should indicate if the user has played that day ❌
  
  
  ## Extra Credit

- Clicking on an item in the daily word stats page should take the player to that game and allow them to play it ❌
- Create another ad page and choose one at random ❌
- Store password in KeyVault ❌

**Pull request closed without merging**